server/src/main/scala/org/broadinstitute/dsp/workbench/welder/server/ObjectService.scala

```diff
@@ -106,12 +106,17 @@ class ObjectService(
       meta <- googleStorageAlg.retrieveAdaptedGcsMetadata(req.localObjectPath, gsPath, traceId)
       res <- meta match {
         case Some(m) =>
-          if (m.lastLockedBy.isDefined && m.lastLockedBy.get == config.ownerEmail)
-            googleStorageAlg.updateMetadata(gsPath, traceId, metadata).as(AcquireLockResponse.Success)
-          else
-            IO.pure(AcquireLockResponse.LockedByOther)
+          m.lastLockedBy match {
+            case Some(lockedBy) =>
+              if(lockedBy == config.ownerEmail)
+                googleStorageAlg.updateMetadata(gsPath, traceId, metadata).as(AcquireLockResponse.Success)
+              else
+                IO.pure(AcquireLockResponse.LockedByOther)
+            case None =>
+              googleStorageAlg.updateMetadata(gsPath, traceId, metadata).as(AcquireLockResponse.Success)
+          }
         case None =>
-          googleStorageAlg.updateMetadata(gsPath, traceId, metadata).as(AcquireLockResponse.Success)
+          IO.raiseError(NotFoundException(s"${gsPath} not found in Google Storage")) //TODO: is this right?
```

Comment:
i don't think it's necessarily wrong. you shouldn't be able to lock objects that aren't covered by storage links and if it's covered by a storage link it should be getting delocalized to gcs. have we seen this in testing? 

an alternative could be to delocalize it with the lock metadata if it's covered by a storage link.

Reply:
Is there a case where we are actually trying to acquire a lock for a non-existent file? Wouldn't this indicate a bug?

Reply:
file here will always be covered by storagelinks cuz `findStorageLink` before this step.

> Is there a case where we are actually trying to acquire a lock for a non-existent file? Wouldn't this indicate a bug?
What about the first time a file gets delocalized, do we need to have a lock to do that? (maybe current implementation is correct since in safeDelocalize we don't check lock; hence there's no need to acquireLock before safeDelocalize)